Lib/argparse.py

```diff
@@ -1897,7 +1897,9 @@ def consume_optional(start_index):
                     # arguments, try to parse more single-dash options out
                     # of the tail of the option string
                     chars = self.prefix_chars
-                    if arg_count == 0 and option_string[1] not in chars:
+                    if (arg_count == 0 and
+                        option_string[1] not in chars and
+                        explicit_arg != ''):
```

Comment:
Optional style nit (increases visual separation between if condition and body)
```suggestion
                    if (
                        arg_count == 0
                        and option_string[1] not in chars
                        and explicit_arg != ''
                    ):
```